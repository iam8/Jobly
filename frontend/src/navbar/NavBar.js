import React, {useContext} from "react";
import { NavLink, Link } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

import UserContext from "../auth/UserContext";
import "./NavBar.css";


/**
 * Render a navigation bar for the site. Appears at the top of every page.
 *  - When user is logged in: displays links to main pages of the site.
 *  - When user is logged out: displays links only to login and signup forms.
 *
 * Props:
 *  - logout(): log out the current user; passed by parent
 */
function NavBar({logout}) {
    const {currentUser} = useContext(UserContext);

    /** Logged-out appearance for navbar. */
    function loggedOutNavbar() {
        return (
            <>
                <NavItem>
                    <NavLink to="/signup">Sign up</NavLink>
                </NavItem>

                <NavItem>
                    <NavLink to="/login">Log in</NavLink>
                </NavItem>
            </>
        );
    }

    /** Logged-in appearance for navbar. */
    function loggedInNavbar() {
        return (
            <>
                <NavItem>
                    <NavLink to="/companies">Companies</NavLink>
                </NavItem>

                <NavItem>
                    <NavLink to="/jobs">Jobs</NavLink>
                </NavItem>

                <NavItem>
                    <NavLink to="/profile">Profile ({currentUser.username})</NavLink>
                </NavItem>

                <NavItem>
                    <Link to="/" onClick={logout}>Log out</Link>
                </NavItem>
            </>
        );
    }

    return (
        <div className="NavBar">
            <Navbar expand="md">
                <NavLink exact to="/" className="navbar-brand">
                    Jobly
                </NavLink>

                <Nav className="ml-auto">
                    {currentUser ? loggedInNavbar() : loggedOutNavbar()}
                </Nav>
            </Navbar>
        </div>
    )
}


export default NavBar;
