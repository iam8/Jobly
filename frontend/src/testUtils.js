import React from "react";
import UserContext from "./auth/UserContext";


const demoUser = {
    username: "testuser",
    firstName: "testfirst",
    lastName: "testlast",
    email: "test@test.net",
};


const DemoUserProvider = ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
        {children}
    </UserContext.Provider>
);


export { DemoUserProvider };
