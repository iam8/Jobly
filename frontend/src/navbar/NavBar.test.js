import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import { DemoUserProvider as UserProvider } from "../testUtils";


test("Renders without crashing", () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );
});


test("Matches snapshot when logged in", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});


test("Matches snapshot when logged out", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});
