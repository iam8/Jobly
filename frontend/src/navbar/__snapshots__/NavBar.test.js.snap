// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile (testuser)
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/"
            >
              Log out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <nav
      class="navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          aria-current="page"
          class="navbar-brand active"
          href="/"
        >
          Jobly
        </a>
        <ul
          class="ml-auto nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/signup"
            >
              Sign up
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Log in
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
